app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -168,6 +170,20 @@ open class HomeActivity : AppCompatActivity(), ShareFragment.TabsSharedCallback
         }
     }
 
+    /**
+     * External sources such as 3rd party links and shortcuts use this function to enter
+     * private mode directly before the content view is created.
+     */
+    private fun setPrivateMode() {
```

Comment:
nit: consider renaming this to `setPrivateModeIfNecessary` since it is not guaranteed that the user will be in private mode after this function resolves.